lib/reporter.js

```diff
@@ -234,8 +236,17 @@ var CoverageReporter = function (rootConfig, helper, logger, emitter, executor)
     // we use these for sending the coverage information outside the scope of a test or run.
     executor.socketIoSockets.on('connection', function (socket) {
       socket.on('coverage', function(info) {
+        coverageResults++;
```

Comment:
I didn't comment on the previous PR, but looking at it today, I think line 222 of this file (hidden above) should be moved into the conditional block ending at line 221. The current logic is appending a non-coverage value in the collector... Basically checking for the string type verifies that the parse call won't explode, but appending the unexpected value to the collector is undoubtedly going to cause a problem later. Either fail right away, or avoid failure.

Reply:
I thought that logic was turning string coverage results into objects, but otherwise it is using an object directly.  That is, socket.io sending from client socket.emit('coverage', { 'a': blah}) will be received as an object not a string. We always send strings, so I think you are right there is no reason to handle the non-string case, but isn't this what it was doing? FYI this was already being done this way prior to my change, I moved it into a helper function though unless I am misunderstanding something:
https://github.com/HBOCodeLabs/karma-coverage/blob/76c2ee53f6925a57bad3a84aab06d2556ca353a9/lib/reporter.js#L234
```
    if (typeof result.coverage === 'string') {
      result.coverage = JSON.parse(result.coverage);
    }

    collector.add(result.coverage)
```

Reply:
I didn't pay a ton of attention to the old code. Looking again, you're right, you just moved it. However, you added a comment, and the comment says that the coverage is not added if it's not defined, and that isn't true. So maybe just remove the false parts of the commentary?